src/main/java/seedu/address/logic/parser/ParserUtil.java

```diff
@@ -95,6 +97,24 @@ public static Email parseEmail(String email) throws ParseException {
         return new Email(trimmedEmail);
     }
 
+    /**
+     * Parses a {@code String gender} into an {@code Gender}.
+     * Leading and trailing whitespaces will be trimmed.
+     *
+     * @throws ParseException if the given {@code gender} is invalid.
+     */
+    public static Gender parseGender(String gender, Boolean isNaAllowed) throws ParseException {
+        requireNonNull(gender);
+        String trimmedGender = gender.trim();
+        if (isNaAllowed && !Gender.isValidGender(trimmedGender)) {
+            throw new ParseException(Gender.MESSAGE_CONSTRAINTS);
+        }
+        if (!isNaAllowed && (!Gender.isValidGender(trimmedGender) || trimmedGender.equals(NA.toString()))) {
+            throw new ParseException(Gender.MESSAGE_CONSTRAINTS);
+        }
+        return new Gender(gender);
```

Comment:
It looks like the logic to check the `isNaAllowed` flag is better suited to be within the `Gender.isValidGender()` method instead. It feels like this is breaking abstraction since `parseGender()` has to determine what is a valid gender instead of leaving that decision to the `isValidGender()` method.

So perhaps move the logic to check `isNaAllowed` to within `Gender.isValidGender()`? Then this `parseGender()` method can just call `Gender.isValidGender()` like the following?

```suggestion
        if (!Gender.isValidGender(trimmedGender)) {
            throw new ParseException(Gender.MESSAGE_CONSTRAINTS);
        }
        return new Gender(gender);
```

Reply:
Thanks! that's a good point, I will change this part